advisor_listener/advisor_listener.py

```diff
@@ -416,11 +399,25 @@ def process_message(msg):
             LOGGER.debug('Skipped advisor result due to missing insights entitlement.')
             return
 
-        LOGGER.info("Evaluating rule hits for inventory_id: %s", msg_dict['input']['host']['id'])
-
         advisor_json, rule_hits = parse_inventory_data(msg_dict)
         if advisor_json:
-            db_import_system(msg_dict, rule_hits, advisor_json, loop)
+            LOGGER.info("Evaluating rule hits for inventory_id: %s", msg_dict['input']['host']['id'])
+            status = db_import_system(msg_dict, rule_hits, advisor_json, loop)
+            if ImportStatus.CHANGED in status:
```

Comment:
Because you removed line 197, if the `db_import_system` function throws exc on call to `db_import_rule_hits`, the flags can be (failed | changed) and in this `if` statement you will log `Finished evaluating rule hits for inventory_id` + send successful message on payload tracker meanwhile you sent `error` message in the `db_import_system` because of the `except DatabaseError` block.

Reply:
Agree, thank you